app/models/observation.rb

```diff
@@ -225,62 +225,42 @@ def self.refresh_content_filter_caches
-    refresh_cached_column_fix_errors(type, foreign, local)
-    msgs
-  end
-
-  # Check how many entries are broken in a mirrored column.  It will be good to
-  # keep track of this at first to make sure we've caught all the ways in which
-  # the mirror can get inadvertently broken.
-  def self.report_broken_caches(type, foreign, local)
-    Observation.connection.select_values(%(
-      SELECT o.id
-      FROM observations o, #{type}s x
-      WHERE o.#{type}_id = x.id
-        AND o.#{local} != x.#{foreign}
-    )).map do |id|
+    tbl = type.camelize.constantize.arel_table
+    broken_caches = Observation.joins(type.to_sym).
+                    where(Observation[local.to_s.to_sym].
+                          not_eq(tbl[foreign.to_s.to_sym]))
+    broken_caches.map do |id|
       "Fixing #{type} #{foreign} for obs ##{id}."
     end
-  end
-
-  # Refresh the mirror of a foreign table's column in the observations table.
-  def self.refresh_cached_column_fix_errors(type, foreign, local)
-    Observation.connection.execute(%(
-      UPDATE observations o, #{type}s x
-      SET o.#{local} = x.#{foreign}
-      WHERE o.#{type}_id = x.id
-        AND o.#{local} != x.#{foreign}
-    ))
+    # Refresh the mirror of a foreign table's column in the observations table.
+    broken_caches.update_all(
+      sanitize_sql_for_assignment(
+        "observations.#{local} = #{type.pluralize}.#{foreign}"
+      )
+    )
   end
 
   # Used by Name and Location to update the observation cache when a cached
   # field value is changed.
   def self.update_cache(type, field, id, val)
-    Observation.connection.execute(%(
-      UPDATE observations
-      SET `#{field}` = #{Observation.connection.quote(val)}
-      WHERE #{type}_id = #{id}
-    ))
+    Observation.where("#{type}_id": id).update_all("#{field}": val)
   end
 
   # Check for any observations whose consensus is a misspelled name.  This can
   # mess up the mirrors because misspelled names are "invisible", so their
   # classification and lifeform and such will not necessarily be kept up to
   # date.  Fixes and returns a messages for each one that was wrong.
   def self.make_sure_no_observations_are_misspelled
-    msgs = Observation.connection.select_rows(%(
-      SELECT o.id, n.text_name FROM observations o, names n
-      WHERE o.name_id = n.id AND n.correct_spelling_id IS NOT NULL
-    )).map do |id, search_name|
+    misspellings = Observation.joins(:name).
+                   where(Name[:correct_spelling_id].not_eq(nil))
+
+    misspellings.pluck(Observation[:id], Name[:text_name]).map do
+      |id, search_name|
       "Observation ##{id} was misspelled: #{search_name.inspect}"
     end
-    Observation.connection.execute(%(
-      UPDATE observations o, names n
-      SET o.name_id = n.correct_spelling_id
-      WHERE o.name_id = n.id AND n.correct_spelling_id IS NOT NULL
-    ))
-    msgs
+    misspellings.update_all(
+      "`observations`.`name_id` = `names`.`correct_spelling_id`"
+    )
```

Comment:
```suggestion
    misspellings.update_all("`name` = `names`.`correct_spelling_id`")
```
Slightly shorter; fits on one line. I don't feel strongly about it.

Reply:
@pellaea yes it is straight-up SQL! I tried all sorts of things to get it to assign without a string, nothing worked... maybe because it is a join? Like: `misspellings.update_all(name_id: Name[:correct_spelling_id])`

@JoeCohen - I believe as above this needs to be `name_id` since that's the column name. Could be something (many things!) I don't know about it though.